### ISSUE TITLE
Extreme latency when typing into a block

### ISSUE THREAD

**Issue by commorancy** [WordPress/gutenberg#24210]
**Describe the bug**
This is a performance issue. While this may have been reported before, I couldn't find anything similar. It could be related to [#21265](https://github.com/WordPress/gutenberg/issues/21265).

Typing into certain block types is exceedingly slow, perhaps up to 1 second lag from the time I press the keyboard buttons to form a word until the time the word appears on screen. It gets worse the faster I type. For example, typing this ticket into Github's editor is real-time. There is no issue with me typing this specific report into Github's editor and it does not display any lag between the keyboard and Github's editor. However, in Gutenberg, this long catchup lag is very, very real... to the point of making the editor unusable.

This seems to affect all text block types to some degree, but most especially affects bullet list blocks. It may have something to do with the length of the block or of the overall article or some combination between the two. My articles sometimes reach 8k to 10k words. In this article I'm writing now, a bulleted section might have upwards of 500 words.

The lag in the bullet list actually seems to grow worse whenever I'm editing the middle of an article and the editor is having to recalculate and move the text below where I'm typing. The faster I type, the worse it seems to get. I can type around 70-85 words per minute or about 330-400 CPM (which isn't as fast as some typists).

**To reproduce**
1. Create a numbered bullet list
2. Begin typing into that list
3. Create several longish paragraphs under several numbers
4. Input slows WAY down to the point that I'm no longer typing in real-time into the editor.

**Expected behavior**
Typing into blocks should remain real-time. I do not experience this behavior on any other editor other than Gutenberg. It's so painful to type into this editor that I have recently considered moving to a different editor. This is definitely a performance issue that needs to be quickly addressed.

**Screenshots**
There is no way to screen shot this or even take a video. There's no way to demonstrate the difference in timing from when I physically press a keyboard button to the time the character appears on screen in the block. Suffice it to say that it's hundreds of milliseconds... perhaps even up to 1 second latency.

**Editor version (please complete the following information):**
- WordPress version: See Wordpress.com
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? See Wordpress.com
- If the Gutenberg plugin is installed, which version is it? See Wordpress.com

**Desktop (please complete the following information):**
 - OS: MacOS X Mojave 10.14.6
 - Browser Firefox 
 - Browser Version 78.0.2

**Comment by ellatrix** [comment]
Would you say that this issue is specific to lists?

**Comment by commorancy** [comment]
Yes, it seems more problematic with bulleted lists than any other text input block type. Though, I have experienced some latency (limited) even when using standard paragraph blocks. When I'm just starting out with a new article, it's far less of a problem than when I am working on a 3000 word (or longer) article and then trying to insert new blocks in the middle. A recently published article has a lot of bullets with long explanations within each bullet. It was painful to edit these sections in this way. I kept fumbling to type because I couldn't see what was showing up until much later... which makes it much harder to correct words which are mistyped.

In general, though, it's much slower to type in Gutenberg overall than it is in Calypso. It seems all of the extra meta-handling around the blocks may be getting in the way of performance. Though, that's just a guess.
